### ISSUE TITLE
Should we rename `static_push` to `push_path`?

### ISSUE THREAD

**Issue by josevalim** [phoenixframework/phoenix#2849]
All of the functions in the Routes/Helpers module end with `url` or `path`. This breaks this standard. The explicit alias helps it but we are still breaking the convention.

The only downside for using `push_path` is that it may conflict with a PushController.

It may also be worth considering if we are doing to have `static_hint` or `hint_path` for early hints. It is also worth considering we can hint/push anything, even dynamically generated assets.

**Comment by josevalim** [comment]
@wojtekmach unfortunately we need the connection, so it would be `push conn, Routes.static_path(conn, path)`. It could work, but I would prefer something a bit cleaner.

**Comment by Qqwy** [comment]
@josevalim Unless we want to be mean wizards and use some macro-magic to steal the `conn` argument before it is passed to the inner function.:angel:  But let's not go there :smiling_imp: .

What about `static_path_autopush(...)`? Then it still has the same explicitness as before, and the `*_autopush` postfix clearly indicates that what is happening is an optimization on what happens without it. Similar to how `start_link` is 'start but then with someting extra' or how `Map.pop_lazy` is just like 'pop' but then lazy.

**Comment by Qqwy** [comment]
While continuing on too long about this can of course be considered bikeshedding, I'd like to know what happens now. Was this issue closed because:

- a decision has been reached? (If so, I'd like to know which one :sweat_smile: )

or

- it was not important and was inactive for too long?

**Comment by josevalim** [comment]
Oops, I wrote the reason but forgot to comment it.

We are not adding those functions anymore because the functionality is still inconsistent and tricky to implement properly. In any case, we decided to go with static_push because if we add it as push_path, we would need to write it as `push_path(static_path("app.css"))` for regular assets. But anyone can still write their own helpers that call `Plug.Conn.push/2`.
